v4/adw/adw-tab-overview.go

```diff
@@ -430,6 +461,24 @@ func (x *TabOverview) ConnectExtraDragValue(cb func(TabOverview, uintptr, uintpt
 	return gobject.SignalConnect(x.GoPointer(), "extra-drag-value", purego.NewCallback(fcb))
 }
 
+// This signal is emitted when the dropped content is preloaded.
+//
+// In order for data to be preloaded, [property@TabOverview:extra-drag-preload]
+// must be set to `TRUE`.
+//
+// The content must be of one of the types set up via
+// [method@TabOverview.setup_extra_drop_target].
+//
+// See [property@Gtk.DropTarget:value].
+//
+// The cb parameter must be provided by allocation of a callback via gobject.NewCallback().
+// The wrapped callback must have the function signature:
+//
+// func(clsPtr uintptr, PageVarp uintptr, ValueVarp uintptr) gdk.DragAction {}
```

Comment:
The thing I dislike a bit about this approach is that the user has no idea how to construct this function, I think we need an intermediate helper that does the same as 
```go
	fcb := func(clsPtr uintptr, PageVarp uintptr, ValueVarp uintptr) bool {
		fa := TabOverview{}
		fa.Ptr = clsPtr

		return cb(fa, PageVarp, ValueVarp)

	}
```

maybe
```go

func GenExtraDragValueCbArg(cb func(TabOverview, uintptr, uintptr) gdk.DragAction) uintptr {
         return gobject.NewCallback(func(clsPtr uintptr, PageVarp uintptr, ValueVarp uintptr) gdk.DragAction {
	        fa := TabOverview{}
	        fa.Ptr = clsPtr

	        return cb(fa, PageVarp, ValueVarp)
         }) 
}
```

Reply:
Another approach:

Would it maybe work if gobject.NewCallback remembers which functions it has created and return the uinptr? We can check function equality using reflect.ValueOf:

https://go.dev/play/p/2lOxzqlKShZ

that means the user has to store the function in a var

Reply:
This is an interesting idea. Use of reflect could be avoided if the user was required to pass in a function pointer instead, then we could just take the address of the pointer and compare to stored values.

This could be made to work for wrapped values too though - we would store the wrapped callback ptr in a map keyed by the func val/ptr, and check the incoming func ptr against known values before creating the wrapper:

ptr example:
https://go.dev/play/p/Ikj2TjyW8N-

reflect example:
https://go.dev/play/p/8vWQjdljjhb

Obviously we'd wrap the cache up in gobject or somewhere and do locking etc.

The pointer version probably makes it more clear that you should pass a value, but would still require a separate api for compat. I don't necessarily hate the idea that in the reflect version simply passing a variable causes the value to be reused, but would need some docs to make it obvious why you should do this.